packages/lib/src/vis/shared/Annotation.tsx

```diff
@@ -1,28 +1,39 @@
+import { useThree } from '@react-three/fiber';
 import type { CSSProperties, HTMLAttributes } from 'react';
+import { Vector2 } from 'three';
 
+import { useWorldToHtml } from '../hooks';
 import { useAxisSystemContext } from './AxisSystemContext';
 import Overlay from './Overlay';
 
 interface Props extends HTMLAttributes<HTMLDivElement> {
   x: number;
   y: number;
-  z?: number;
   scaleOnZoom?: boolean;
   style?: CSSProperties;
 }
 
 function Annotation(props: Props) {
-  const { x, y, z = 1, scaleOnZoom, style, children, ...divProps } = props;
-  const { abscissaScale, ordinateScale } = useAxisSystemContext();
+  const { x, y, scaleOnZoom, children, ...divProps } = props;
+
+  const camera = useThree((state) => state.camera);
+
+  const { dataToWorld } = useAxisSystemContext();
+  const worldToHtml = useWorldToHtml();
+
+  const htmlPt = worldToHtml(dataToWorld(new Vector2(x, y)));
 
   return (
     <Overlay
```

Comment:
I think you could use `Html` directly, and just set `position: absolute` and `pointer-events: none` yourself (and move those styles to the inner `div`). This way, the overlay is really just to put a full-size `div` on top of the canvas (or another element).